core/tests/protractor_utils/ExplorationEditorTranslationTab.js

```diff
@@ -165,6 +167,97 @@ var ExplorationEditorTranslationTab = function() {
       element(by.cssContainingText('option', language)).click();
   };
 
+  var selectedLanguageElement = element(
+    by.css('.protractor-test-translation-language-selector')).element(
+    by.css('option:checked'));
+
+  var languageSelectorLabelElement = element(
+    by.css('.protractor-test-language-selector-label'));
+  var progressBarLabelElement = element(
+    by.css('.protractor-test-progress-info'));
+  var translationModeButton = element(
+    by.css('.protractor-test-translation-mode'));
+  var voiceoverModeButton = element(by.css('.protractor-test-voiceover-mode'));
+
+  var saveTranslationButton = element(
+    by.css('.protractor-test-save-translation'));
+  var editTranslation = element(
+    by.css('.protractor-test-edit-translation'));
+  var translationDisplay = element(
+    by.css('.protractor-test-translation-display'));
+
+  var stateGraph = element(by.css('.protractor-test-translation-graph'));
+  var stateBackgroundNodes = stateGraph.all(by.css(
+    '.protractor-test-node-background'));
+  var stateNodes = stateGraph.all(by.css('.protractor-test-node'));
+  var stateNodeLabel = function(nodeElement) {
+    return nodeElement.element(by.css('.protractor-test-node-label'));
+  };
+
+  this.setTranslation = function(richTextInstructions) {
+    waitFor.elementToBeClickable(
+      editTranslation,
+      'editTranslation taking too long to appear to set content');
+    editTranslation.click();
+    var stateTranslationEditorTag = element(
+      by.tagName('state-translation-editor'));
+    var stateTranslationEditor = stateTranslationEditorTag.element(
+      by.css('.protractor-test-state-translation-editor'));
+    waitFor.visibilityOf(
+      stateTranslationEditor,
+      'stateTranslationEditor taking too long to appear to set content');
+    var richTextEditor = forms.RichTextEditor(stateTranslationEditor);
+    richTextEditor.clear();
+    richTextInstructions(richTextEditor);
+    expect(saveTranslationButton.isDisplayed()).toBe(true);
+    saveTranslationButton.click();
+    waitFor.invisibilityOf(
+      saveTranslationButton,
+      'State translation editor takes too long to disappear');
+  };
+
+  this.expectTranslationToMatch = function(richTextInstructions) {
+    forms.expectRichText(translationDisplay).toMatch(richTextInstructions);
+  };
+
+  this.switchToVoiceoverMode = function() {
+    waitFor.elementToBeClickable(
+      voiceoverModeButton,
+      'Translation Mode switch is taking too long to appear');
```

Comment:
Why are we waiting for voiceover mode button but the error message says Translation?

Reply:
Oops, I'll fix this!

Reply:
Done!